MobileCenterDistribute/MobileCenterDistribute/MSDistribute.m

```diff
@@ -679,14 +679,19 @@ - (void)closeApp __attribute__((noreturn)) {
   exit(0);
 }
 
-- (void)openUrl:(NSURL *)url {
+- (BOOL)openUrl:(NSURL *)url {
+  
+  /*
+   * Ignore if app secret not set, can't test scheme.
+   * Also ignore if request is not for Mobile Center Distribute and this app.
+   */
+  if (!self.appSecret || ![[url scheme] isEqualToString:[NSString stringWithFormat:kMSDefaultCustomSchemeFormat, self.appSecret]]) {
```

Comment:
We already know the app secret so the SDK can determine the exact URL scheme for the app. In this case, we can also log to console if developers don't specify it properly so it would be better than just checking the prefix.